noty-api/application/src/dev/shreyaspatil/noty/api/controller/NotesController.kt

```diff
@@ -97,6 +98,21 @@ class NotesController @Inject constructor(private val noteDao: NoteDao) {
         }
     }
 
+    fun pinNote(user: User, noteId: String, pinRequest: PinRequest): NoteResponse {
+        return try {
+            checkNoteExistsOrThrowException(noteId)
+            checkOwnerOrThrowException(user.id, noteId)
+            val id = noteDao.pinById(noteId, pinRequest.isPinned)
```

Comment:
Also, DAO method can be renamed. Like, `updateNotePinById` or something like this.

Reply:
Renamed it as per suggestion.